lib/omniauth/strategies/openid_connect.rb

```diff
@@ -464,9 +465,14 @@ def configured_response_type
       def verify_id_token!(id_token)
         return unless id_token
 
-        decode_id_token(id_token).verify!(issuer: options.issuer,
-                                          client_id: client_options.identifier,
-                                          nonce: params['nonce'].presence || stored_nonce)
+        verify_kwargs = {
+          issuer: options.issuer,
+          client_id: client_options.identifier,
+          nonce: params['nonce'].presence || stored_nonce
+        }
+        verify_kwargs[:audience] = client_options.audience if client_options.audience.present?
```

Comment:
I think `present?` is only available with ActiveSupport:

```suggestion
        verify_kwargs[:audience] = client_options.audience unless client_options.audience.to_s.empty?
```

Reply:
Or maybe it suffices to do this:

```suggestion
        verify_kwargs[:audience] = client_options.audience if client_options.audience
```

Reply:
done!

Reply:
I'm surprised Rubocop didn't complain about the use of `merge!` for a single item, but 🤷